tokio-net/src/udp/socket.rs

```diff
@@ -141,8 +176,16 @@ impl UdpSocket {
     ///
     /// The future will resolve to an error if the IP version of the socket does
     /// not match that of `target`.
-    pub async fn send_to(&mut self, buf: &[u8], target: &SocketAddr) -> io::Result<usize> {
-        poll_fn(|cx| self.poll_send_to_priv(cx, buf, target)).await
+    pub async fn send_to<A: ToSocketAddrs>(&mut self, buf: &[u8], target: A) -> io::Result<usize> {
```

Comment:
I don't think it is good idea for send to always resolve target's addr.

Most likely it would be preferable for user to have option to use already resolved SocketAddr

At least as separate API

Reply:
indeed aside from extra allocation :)

Well it is up to you. It is just some APIs have another API that accept ScoketAddr directly,
Should be aligned?

Reply:
I meant extra allocation when user pass `SocketAddr` if `ToSocketAddr` would impl for it like in std

Right now there `bind_addr`, `connect_addr` which take `SocketAddr`.
But not for `send_to` :)